[16.0][REF] pingen_env: Leverage server-env mixins

grindtildeath opened the pull request.

simahawk commented on pingen_env/models/res_company.py:
```diff
@@ -1,46 +1,22 @@
-        compute="_compute_pingen_env", required=False, readonly=True
-    )
+    _name = "res.company"
+    _inherit = ["res.company", "server.env.mixin", "server.env.techname.mixin"]
```
FTR using the techname mixin implies that a techname is set on existing records and the env configuration must be updated. As a fallback, I would say that we need a migration script to automatically set the techname = to the current name. Tech name is normalized, so in the script it should be set via SQL if the original name is not normalized.

@grindtildeath am I wrong?

grindtildeath replied:
You're right on the techname, but there's no way a migration script is enough to have a "transparent" migration, as server environment values will have to be manually updated to reflect the techname anyway.

I'm not sure what's the best way to handle that though...

simahawk replied:
I bet so far it relied on `name`. Just set the techname = name by default and everything should work fine.
Yet, as I mentioned, you should use an sql to update the name and bypass the normalization.
This way, you can avoid to touch the env conf. At least on the 1st upgrade.

grindtildeath replied:
Actually this will still use the `name` field as `_server_env_section_name_field` :confused:

grindtildeath commented:
cf https://github.com/OCA/server-env/pull/232

grindtildeath commented:
I added a migration script to set the tech_name = name, in case it's not defined through another module yet. I also added a warning in the logs.

@simahawk Can we please merge the server-env PR and this one? :pray:

grindtildeath commented:
ping @simahawk

simahawk commented:
/ocabot merge nobump
